ReactCommon/react/utils/ManagedObjectWrapper.h

```diff
@@ -10,7 +10,7 @@
 #include <react/debug/react_native_assert.h>
 
 #if defined(__OBJC__) && defined(__cplusplus)
-#if TARGET_OS_MAC && TARGET_OS_IPHONE
```

Comment:
Was this a way to target catalyst before? When would both of those be true ever?

Reply:
TARGET_OS_MAC = any Apple OS (macOS, iOS)
TARGET_OS_IPHONE = any iPhone OS (iOS, tv, watch, bridge)

I am not sure why the original line was 'just'
TARGET_OS_IPHONE
as the framework is for iOS apps ...

Meta we agreed this gating is useless, hence I have removed it

See also: https://stackoverflow.com/a/49560690